src/main/io/gps.c

```diff
@@ -2591,13 +2592,24 @@ void GPS_calculateDistanceAndDirectionToHome(void)
     }
 }
 
+void GPS_distances(const gpsLocation_t *from, const gpsLocation_t *to, float *pEWDist, float *pNSDist) {
```

Comment:
converting from spherical to (locally) cartesian difference IS calculating of distance

Reply:
maybe call like geodesianToCartesian, or simmilar.
The distance is not full correctly on my opinion

Reply:
I used the same 'distance' calculation in `GPS_distance_cm_bearing` where the comment is `// Get distance between two points in cm`.

In practice, for the kind of distances we are talking about, the returned value is like any other distance between two points on the Earth's surface at the same altitude, though it assumes a perfect sphere.

Perhaps I can change the earlier comments from
```
// Get distance between two points in cm
// Get bearing from pos1 to pos2, returns an 1deg = 100 precision
```
to
```
// Get distance between two points in cm using spherical to Cartesian transform
// One one latitude unit, or one longitude unit at the equator, is 1.113195 cm.
// Get bearing from pos1 to pos2, returns values with 0.01 degree precision
```

Reply:
`GPS_distances()` doesn't calculate the magnitude of the vector, so it's not distance in the way it is in `GPS_distance_cm_bearing()`. This right here is the "component-wise difference" between two 2D points (also converted into cartesian). That's why I am a bit unsure about the name "GPS_distances".

The code is duplicated but I guess we could clean it up in a refactoring PR.

But I really like the comment suggestion

Reply:
@KarateBrot  I use exactly the same method in `GPS_distances()` as in `GPS_distance_cm_bearing()`...

In `GPS_distance_cm_bearing()`:
```
    float dLat = (to->lat - from->lat) * EARTH_ANGLE_TO_CM;
    float dLon = (to->lon - from->lon) * GPS_cosLat * EARTH_ANGLE_TO_CM; // convert to local angle
    float dAlt = dist3d ? to->altCm - from->altCm : 0; // ignore this because we only care about 2D distances
    *pDist = sqrtf(sq(dLat) + sq(dLon) + sq(dAlt)); 
```

In `GPS_distances()`:
```
    *pNSDist = (float)(to->lat - from->lat) * EARTH_ANGLE_TO_CM;
    *pEWDist = (float)(to->lon - from->lon) * GPS_cosLat * EARTH_ANGLE_TO_CM;
```

Both functions use identical maths to return identical values, eg, dLat = pNSDist, and dLon = pEWDist.

All I need in the autopilot code are the latitude and longitude vector components of the distance between the two points, and they are distances.  I don't use the actual distance length.  However...

Where the `GPS_distance_cm_bearing()` function returns 'distance', it just uses root sum of squares of those vectors.

In autopilot.c `positionControl()`, it's nice to log the absolute distance from target.  So I have vector2_t  `distanceCm = vector2Norm(&gpsDistance)`, where &gpsDistance) contains *pNSDist and *pEWDist.  Then vector2Norm does is a root sum of squares. This Maths returns exactly the same number for 'distance
 as does `GPS_distance_cm_bearing()`.

Hence I still don't quite your comment.

I need a separate function because `GPS_distance_cm_bearing()` doesn't return those vector components, only the distance itself.

Maybe I should re-name `GPS_distances` to `GPS_Lat_Lon_Vectors` to resolve the confusion?

The comment will go in.